tests/suite/test_transport_server_tcp_load_balance.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+import re
+import time
+import socket
+
+from settings import TEST_DATA
+from suite.resources_utils import (
+    wait_before_test,
+    patch_deployment_from_yaml,
+    create_items_from_yaml,
+    get_ingress_nginx_template_conf,
+    get_ts_nginx_template_conf,
+    create_service_from_yaml,
+    delete_service,
+    scale_deployment
+)
+from suite.custom_resources_utils import (
+    patch_ts
+)
+
+
+@pytest.mark.ts
+@pytest.mark.parametrize(
+    "crd_ingress_controller, transport_server_setup",
+    [
+        (
+            {
+                "type": "complete",
+                "extra_args":
+                    [
+                        "-global-configuration=nginx-ingress/nginx-configuration",
+                        "-enable-leader-election=false"
+                    ]
+            },
+            {"example": "transport-server-tcp-load-balance", "app_type": "simple"},
+        )
+    ],
+    indirect=True,
+)
+class TestTransportServerTcpLoadBalance:
+
+    def restore_transport_server(self, kube_apis, transport_server_setup) -> None:
+        """
+        Function to revert a TransportServer resource to a valid state.
+        """
+        transport_server_file = f"{TEST_DATA}/transport-server-status/standard/transport-server.yaml"
+        patch_ts(kube_apis.custom_objects, transport_server_setup.name, transport_server_file, transport_server_setup.namespace)
+
+    @pytest.mark.sean
+    def test_number_of_replicas(
+        self, kube_apis, crd_ingress_controller, transport_server_setup, ingress_controller_prerequisites
+    ):
+        """
+        The load balancing of TCP should result in 4 servers to match the 4 replicas of a service.
+        """
+        original = scale_deployment(kube_apis.apps_v1_api, "tcp-service", transport_server_setup.namespace, 4)
+
+        wait_before_test()
```

Comment:
because we asked k8s to bring up new pods, better to wait until the new pods are running, rather than simply sleeping. I worry that 3 seconds  (the default in wait_before_test) will not be enough all the time for k8s to bring up the new pods.  note that we have `wait_until_all_pods_are_ready` in `resource_utils` to check if the pods are running

Reply:
Great suggestion, thanks!

Reply:
I've tried this but it doesn't work I'm afraid.